.github/workflows/CI.yml

```diff
@@ -17,36 +17,42 @@ env:
   CARGO_PROFILE_RELEASE_CODEGEN_UNITS: 1
 
 jobs:
+  sanity:
+    timeout-minutes: 30
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+        with:
+          prefix-key: ${{ runner.os }}
+      - name: Check format
+        run: cargo fmt-amaru
+      - name: Run clippy
+        run: cargo clippy-amaru
   build:
     name: Build on ${{ matrix.environments.runner }} with target ${{ matrix.environments.target }}
+    needs: sanity
```

Comment:
Or not? If we're really after a speed up, we could have both happen in parallel; if the `sanity` fails, it'll cancel the build anyway and there's in fact no artifact produced by the `sanity` that is needed for the build & test to start.

Reply:
Makes sense!